mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1553,8 +1714,29 @@ createCPUConvWithMLIR(ModuleOp module, func::FuncOp &func,
   affine::buildAffineLoopNest(b, loc, lowerBounds, upperBounds, steps,
                               createConv2dLoopNest);
 
+  if (!opd1.dyn_cast<BlockArgument>())
+    b.create<memref::DeallocOp>(loc, opd1);
+  if (!opd2.dyn_cast<BlockArgument>())
+    b.create<memref::DeallocOp>(loc, opd2);
+  if (!result.dyn_cast<BlockArgument>()) {
+    BlockArgument resultBlockArg;
+    switch (genConfig.operation.value()) {
+    case rock::ConvOpType::Fwd:
+      resultBlockArg = block->getArgument(2);
+      break;
+    case rock::ConvOpType::BwdWeight:
+      resultBlockArg = block->getArgument(0);
+      break;
+    case rock::ConvOpType::BwdData:
+      resultBlockArg = block->getArgument(1);
+      break;
+    }
+
+    Value resultFlat = makeNDMemRef(b, result, 1);
+    emitMemcpy(b, resultFlat, resultBlockArg);
```

Comment:
Doesn't the result need to be free'd?

Reply:
Yup.  In several other cases, too.  Changes coming.